### PR TITLE
InteractiveViewer pan axis locking

### LINKED ISSUE #58043: InteractiveViewer scroll direction locking
InteractiveViewer, introduced in https://github.com/flutter/flutter/pull/56409, should allow the scroll direction to be locked to the x or y axis for any given scroll.  This is similar to the behavior of Google Sheets.  This is also mentioned in https://github.com/flutter/flutter/issues/12878.

### PR DESCRIPTION
**Pull request by justinmc**
## Description

This PR adds an `alignPanAxis` parameter to InteractiveViewer.  When true, panning is only allowed along the x or y axes, not diagonally.

align_pan_axis

## Related Issues

Closes https://github.com/flutter/flutter/issues/58043
Mentioned in https://github.com/flutter/flutter/issues/12878

## Tests

I added the following tests:

  * A 45 degree drag pans vertically.
  * A <45 degree drag pans horizontally.
  * Zoom still allows updating the translation along a diagonal.

## Breaking Change

None.

### CODE REVIEW

File: packages/flutter/lib/src/widgets/interactive_viewer.dart [hunk 1]
```diff
@@ -85,6 +87,12 @@ class InteractiveViewer extends StatefulWidget {
            && boundaryMargin.left.isFinite)),
        super(key: key);
 
+  /// If true, panning is only allowed in the direction of the main axes.
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/widgets/interactive_viewer.dart:90]: nit: I couldn't find the definition of "main axes". Maybe "horizontal axis or vertical axis"?

File: packages/flutter/lib/src/widgets/interactive_viewer.dart [hunk 2]
```diff
@@ -85,6 +87,12 @@ class InteractiveViewer extends StatefulWidget {
...
+  ///
+  /// In other words, when this is true, diagonal panning is not allowed. This
+  /// is a common pattern in tables where data is displayed in columns and rows.
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/widgets/interactive_viewer.dart:93]: It appears to me that if the user initiates a horizontal drag, then they won't be able to change the direction to vertical later (in the same gesture). Maybe add documentation for this behavior too?

File: packages/flutter/lib/src/widgets/interactive_viewer.dart [hunk 3]
```diff
@@ -528,9 +537,14 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
       return matrix.clone();
     }
 
+    final Offset alignedTranslation = widget.alignPanAxis && _panAxis != null
+      && _gestureType == _GestureType.pan
```
Comment (LongCatIsLooong) [inline packages/flutter/lib/src/widgets/interactive_viewer.dart:541]: Can this be done in `_onScaleUpdate` to avoid checking `_gestureType`?

**Review by LongCatIsLooong — APPROVED** [review]

File: packages/flutter/lib/src/widgets/interactive_viewer.dart (hunk 1 above)
Comment (justinmc) [reply packages/flutter/lib/src/widgets/interactive_viewer.dart:90]: Will do, that sounds better.

File: packages/flutter/lib/src/widgets/interactive_viewer.dart (hunk 2 above)
Comment (justinmc) [reply packages/flutter/lib/src/widgets/interactive_viewer.dart:93]: That's right.  I'll document it.  It seems to be how Google Sheets works on my iPhone too.

File: packages/flutter/lib/src/widgets/interactive_viewer.dart (hunk 3 above)
Comment (justinmc) [reply packages/flutter/lib/src/widgets/interactive_viewer.dart]: Good call 👍

**Pull request merged**